core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -63,10 +61,7 @@ public TableScanStatsRule(Metadata metadata, StatsNormalizer normalizer)
             return node.getStatistics();
         }
 
-        // TODO Construct predicate like AddExchanges's LayoutConstraintEvaluator
```

Comment:
My bad, i forgot about this. `applyFilter` won't make this obsolete, and this is a good idea.

however, the current state is bad: we pass a Constraint and it's useless. So i think it's better to proceed with this PR, and reintroduce Constraint later, if actually going to be used.